### PR TITLE
Contact's page

### PR DESCRIPTION
**Pull request by UgurDenli**

### CODE REVIEW

File: WhatsApp/app/build.gradle [hunk 1]
```diff
@@ -55,6 +56,7 @@ dependencies {
     implementation 'io.reactivex.rxjava3:rxjava:3.0.0'
     // Dependency Injection - Hilt
     implementation("com.google.dagger:hilt-android:2.38.1")
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
Comment (asos-oskar) [inline WhatsApp/app/build.gradle:59]: What is this library responsible for?

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/contacts/ContactsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        setHasOptionsMenu(true)
+        super.onCreate(savedInstanceState)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_contacts, container, false)
```
Comment (asos-oskar) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/contacts/ContactsFragment.kt:27]: set up view binding and also inflate the Fragment inside of the Fragment() class on line 15.

Fragment(R.layout.fragment_contacts)

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/home/HomeFragment.kt [hunk 3]
```diff
@@ -21,6 +25,15 @@ class HomeFragment : BaseFragment<FragmentHomeBinding>() {
     override fun inflateBinding(layoutInflater: LayoutInflater): FragmentHomeBinding {
        return FragmentHomeBinding.inflate(layoutInflater)
     }
+
+    override fun onStart() {
```
Comment (asos-oskar) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/home/HomeFragment.kt:29]: any specific reason why we are setting up the FAB inside of onStart and not onCreateView?

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/home/HomeFragment.kt [hunk 4]
```diff
@@ -21,6 +25,15 @@ class HomeFragment : BaseFragment<FragmentHomeBinding>() {
...
+        super.onStart()
+        val myRv = view?.findViewById(R.id.fabbtn) as FloatingActionButton
+
+        myRv.setOnClickListener {
+            findNavController().navigate(R.id.action_homeFragment_to_contactsFragment)
+        }
```
Comment (asos-oskar) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/home/HomeFragment.kt:35]: I think we should create some xml file that holds these ids

File: WhatsApp/app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -1,12 +1,17 @@
...
+<LinearLayout
+    xmlns:app="http://schemas.android.com/apk/res"
     xmlns:tools="http://schemas.android.com/tools"
     xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
     android:orientation="vertical"
     android:layout_height="match_parent"
     tools:context=".ui.MainActivity">
 
+    <include
```
Comment (asos-oskar) [inline WhatsApp/app/src/main/res/layout/activity_main.xml:11]: Why are we adding a content_main that holds fragment navigation when that is already set up in FragmentContainerView

File: WhatsApp/app/src/main/res/layout/fragment_home.xml [hunk 6]
```diff
@@ -85,7 +80,7 @@
 
 
         <com.google.android.material.floatingactionbutton.FloatingActionButton
-            android:id="@+id/fabNewConversation"
+            android:id="@+id/fabbtn"
```
Comment (asos-oskar) [inline WhatsApp/app/src/main/res/layout/fragment_home.xml:83]: why did we rename this id

**Review by asos-oskar — CHANGES_REQUESTED** [review]

File: WhatsApp/app/src/main/res/layout/fragment_home.xml (hunk 6 above)
Comment (UgurDenli) [inline WhatsApp/app/src/main/res/layout/fragment_home.xml]: felt like it was too long tried to shorten it

File: WhatsApp/app/build.gradle (hunk 1 above)
Comment (OskarLasota) [inline WhatsApp/app/build.gradle]: what is this library responsible for?

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/MainActivity.kt [hunk 7]
```diff
@@ -1,15 +1,42 @@
...
+            )
+        )
+        setupActionBarWithNavController(navController)
+        navView.setupWithNavController(navController)
+
+
     }
+
+    override fun onSupportNavigateUp(): Boolean {
+        return navController.navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
```
Comment (OskarLasota) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/MainActivity.kt]: make this a single expression function

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/contacts/ContactsFragment.kt (hunk 2 above)
Comment (OskarLasota) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/contacts/ContactsFragment.kt]: inflate fragment in line 15 - Fragment(R.layout..)

File: WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/home/HomeFragment.kt (hunk 3 above)
Comment (OskarLasota) [inline WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/home/HomeFragment.kt]: why onStart

File: WhatsApp/app/src/main/res/layout/activity_main.xml (hunk 5 above)
Comment (OskarLasota) [inline WhatsApp/app/src/main/res/layout/activity_main.xml]: we are already using a FragmentContainer

File: WhatsApp/app/src/main/res/layout/fragment_contacts.xml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout
```
Comment (OskarLasota) [inline WhatsApp/app/src/main/res/layout/fragment_contacts.xml]: Does this have to be a framelayout?

**Review by OskarLasota — CHANGES_REQUESTED** [review]

File: WhatsApp/app/src/main/res/layout/fragment_contacts.xml (hunk 8 above)
Comment (UgurDenli) [reply WhatsApp/app/src/main/res/layout/fragment_contacts.xml]: not really no

**Review by UgurDenli — COMMENTED** [review]
Looks good from my side i can see for the future changes aswell

**Review by Josh-Owen — APPROVED** [review]
It requires some work still, but we need to get develop up to date so we will change up the implementation in a future update

**Review by OskarLasota — APPROVED** [review]

**Pull request merged**